src/lib/output/plugins/JavascriptIndexPlugin.ts

```diff
@@ -63,12 +64,15 @@ export class JavascriptIndexPlugin extends RendererComponent {
                 parent = undefined;
             }
 
-            const row: any = {
+            const row: IDocument = {
                 id: rows.length,
                 kind: reflection.kind,
                 name: reflection.name,
                 url: reflection.url,
-                classes: reflection.cssClasses,
+                classes: reflection.cssClasses ?? "",
+                categories: (reflection.categories ?? []).map(
```

Comment:
Also need to check that this works properly with categorizeByGroup set to both true and false.

Reply:
I actually wasn't able to get any kind of reaction out of cateogry stuff. How does one activate the category plugin in the example app? I didn't see any reference to it in in typedoc.json

Reply:
The category plugin is always active - but it will only do something if the documented members use `@category` tags

Reply:
Ah, I see; I guess I didn't notice that. Would it be okay if I sprinkle a few in there to exercise some of this stuff?

Reply:
Yep, go ahead

Reply:
So, it turns out that there already are plenty of "category" tags in the example source; the reason I'm not seeing anything in the `reflection.categories` field here is that it's apparently never written to. It seems the categories stuff are grouped up, with each parent node being a category, and the children being a list of the reflections tagged with that category.

Any idea how I can I fish my way to the category information, here? Or do I need to do some writes while processing the category plugin?

Reply:
That's controlled by the `categorizeByGroup` option - `reflection.groups` has a `ReflectionGroup[]` which can contain categories

Reply:
Yes, I've been experimenting with that; I tried setting it to both `true` and `false` and neither value ever seems to result in `reflection.categories` being anything other than `undefined`, here.

Reply:
If I add this to `CategoryPlugin:170`, it starts working as expected:

```
                child.categories = child.categories ?? [];
                child.categories.push(category);
```

Reply:
Hmm, maybe this is where your suggestion to encode the boost score rather than the cats themselves could come into play--I could write the numbers in the Category plugin, and dodge the crazy recursion.

Reply:
6a41cee5aab213bbe29a49b6616bf442251f0ad0